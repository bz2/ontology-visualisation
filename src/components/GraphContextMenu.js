import { Menu } from 'primereact/menu'
import { connect } from 'redux-zero/react'
import PropTypes from 'prop-types'
import { useTranslation } from 'react-i18next'
import actions from '../store/actions'
import expandNode from '../utils/graphVisualisation/expandNode'
import { SIDEBAR_VIEW_CUSTOM_QUERY } from '../constants/views'

const GraphContextMenu = ({
  setStoreState,
  contextMenuData,
  addNumber,
}) => {
  const { t } = useTranslation()

  const {
    top,
    left,
    nodeId
  } = contextMenuData

  const menu = [
    {
      label: t('customQuery'),
      icon: 'pi pi-fw pi-tablet',
      command: () => {
        setStoreState('sidebarView', SIDEBAR_VIEW_CUSTOM_QUERY)
        setStoreState('showContextMenu', false)
      }
    },
    {
      separator: true
    },
    {
      label: t('close'),
      icon: 'pi pi-fw pi-power-off',
      command: () => setStoreState('showContextMenu', false)
    }
  ]

  if (nodeId) {
    menu[1] = {
      label: t('expandNode'),
      icon: 'pi pi-fw pi-plus',
      command: () => {
        expandNode({
          nodeId,
          setStoreState,
          addNumber,
        })
        setStoreState('showContextMenu', false)
      }
    }
  }

  if (nodeId) {
    menu[2] = {
      label: t('addNote'),
      icon: 'pi pi-fw pi-plus',
      command: () => {
        expandNode({
          nodeId,
          setStoreState,
          addNumber,
        })
        setStoreState('showContextMenu', false)
      }
    }
  }

  return (
    <Menu
      className="context-menu"
      style={{ top, left }}
      model={menu}
    />
  )
}

GraphContextMenu.propTypes = {
  setStoreState: PropTypes.func.isRequired,
  addNumber: PropTypes.func.isRequired,
  contextMenuData: PropTypes.shape().isRequired,
}

const mapToProps = ({
  contextMenuData,
}) => ({
  contextMenuData,
})

export default connect(
  mapToProps,
  actions
)(GraphContextMenu)
