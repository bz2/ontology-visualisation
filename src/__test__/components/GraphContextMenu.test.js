import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import GraphContextMenu from '../../components/GraphContextMenu'

const setup = () => {
  const props = {
    contextMenuData: {
      top: 100,
      left: 100,
      nodeId: 'node-123'
    },
    setStoreState: jest.fn(),
    addNumber: jest.fn(),
  }

  const component = shallow(<GraphContextMenu {...props} />)

  return {
    component,
    props
  }
}

describe('GraphContextMenu', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should match snapshot', () => {
    const {
      component
    } = setup({
      customQueryOutput: undefined
    })

    expect(toJson(component)).toMatchSnapshot()
  })
})
