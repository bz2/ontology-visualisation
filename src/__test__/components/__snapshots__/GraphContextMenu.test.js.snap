// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphContextMenu should match snapshot 1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "resetSelectedEdges": [Function],
      "resetSelectedNodes": [Function],
      "setStoreState": [Function],
      "updateGraphData": [Function],
    }
  }
  contextMenuData={
    Object {
      "left": 100,
      "nodeId": "node-123",
      "top": 100,
    }
  }
  mapToProps={[Function]}
  setStoreState={[MockFunction]}
>
  <Component />
</Connect>
`;
